### PR TITLE
Auth backup7

### PR DESCRIPTION
**Pull request by GonzalezEstevan**
Screen Shot 2022-03-22 at 7 24 00 PM
Screen Shot 2022-03-22 at 7 29 10 PM
Screen Shot 2022-03-22 at 7 29 35 PM

* Added small login and signup edits 
* also added on signup information is sent to the backend so the front end can pull actual information back 
* kept some information in state in case we needed to send it to the front end 
 (lmk if you want me to get rid of it)
 ## Things I need to do still 
 * Add account type for google and facebook sign ins

### CODE REVIEW

File: pages/signup.js [hunk 1]
```diff
@@ -49,21 +64,43 @@ export default function Signup() {
...
+            <Input sx={{ my: .3 }} disableUnderline={true} onChange={(e) => handleSignUpFormInput(e.target.value, 'password')} placeholder="Password" className={classes.userInput} type="password"></Input>
+            <Button sx={{ my: 2 }} type='submit' className={classes.loginButton}>Create Account</Button>
           </form>
           <div className={{ flexDirection: 'column' }}>
-            <Button sx={{ my: 2 }} onClick={signInWithGoogle} className={classes.googleButton} startIcon={<GoogleIcon />}>Continue with Google</Button>
-            <Button onClick={signInWithFacebook} className={classes.facebookButton} startIcon={<FacebookIcon />}>Continue with Facebook</Button>
+            <Button sx={{ my: 2 }} disabled={filledFormFlag} onClick={(e) => signInWithGoogle(signupInfo)} className={classes.googleButton} startIcon={<GoogleIcon />}>Continue with Google</Button>
+            <Button onClick={(e) => signInWithFacebook(signupInfo) } disabled={filledFormFlag} className={classes.facebookButton} startIcon={<FacebookIcon />}>Continue with Facebook</Button>
```
Comment (tiffanyyv) [inline pages/signup.js:97]: Can we use the "Main Button" here?

File: utils/context/AuthProvider.js [hunk 2]
```diff
@@ -18,31 +19,54 @@ const AuthContext = createContext();
...
+          firstName: body.firstName,
+          lastName: body.lastname,
+          username: body.username,
+          location: body.location
+        }
+        console.log(postBody)
+        if (body.account_type === 'Mentor') {
+          axios.post(`http://localhost:3000/api/users`, postBody)
+            .then(response => {
+              console.log(response)
```
Comment (tiffanyyv) [inline utils/context/AuthProvider.js:41]: @cmjosephs do we want to use console.warn here?

File: utils/context/AuthProvider.js [hunk 3]
```diff
@@ -53,22 +77,24 @@ export function AuthProvider({ children }) {
...
   }
-  const signInWithFacebook = () => {
+  const signInWithFacebook = (accountInfoObj) => {
     const provider = new FacebookAuthProvider();
     signInWithPopup(auth, provider)
-      .then(res => {
+      .then(async response => {
+        var response = await fetch(`/api/users/${user.uid}`)
+        var jsonData = await response.json();
         console.log(res)
```
Comment (tiffanyyv) [inline utils/context/AuthProvider.js:98]: Remember to take these console.logs out when you're done :D

**Review by tiffanyyv — COMMENTED** [review]
👯

File: utils/context/AuthProvider.js [hunk 4]
```diff
@@ -18,31 +19,54 @@ const AuthContext = createContext();
...
         router.push('/app/my-courses')
-        console.log(response)
       })
       .catch((err) => {
         console.warn('Problem with sign up: ', err.message);
       })
   }
 
   const login = (email, password) => {
-    console.log(email, password)
```
Comment (cmjosephs) [inline utils/context/AuthProvider.js:36]: Remove during production

**Review by cmjosephs — APPROVED** [review]
😃

File: pages/signup.js (hunk 1 above)
Comment (GonzalezEstevan) [reply pages/signup.js:97]: Main Button?

**Pull request merged**